app/controllers/evss_base_controller.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class EVSSBaseController < ApplicationController
+  before_action :authorize_user
+
+  private
+
+  def authorize_user
+    authorize :evss, :access?
+  end
+end
```

Comment:
This seems thin to warrant it's own class. It's also a bit more obfuscated as opposed to the plain old declaration at the top of a controller:

```ruby
before_action except: [:some_other_action] do
  authorize :evss, :access?
end
```

Finally, what if a controller contains some actions that require authz and some that dont?

Kind of a nitpick, doesn't matter all that much either way.

Reply:
@omgitsbillryan agree that the base evss controller may not be needed anymore. 

> what if a controller contains some actions that require authz and some that dont?

In that case you could use `:only` or `:except` to filter actions, or don't have a before_action and add pundit authorize calls to the actions that need it (this is also how you can apply different policy levels per action):
```ruby
def update
  authorize :evss, :update?
  # only users who pass `update?`
end

def show
  authorize :evss, :read?
  # only users who pass `read?`
end
```

Reply:
I don't think I was clear.  My statement:

> what if a controller contains some actions that require authz and some that dont?

poking at the `before_action :authorize_user` inside of `evss_base_controller.rb` which meant that every action in a subclass of `EVSSBaseController` would be forced to authorize at `access?` even though that might not be desirable.

Reply:
ah ok, for a controller that subclasses the evss base controller that would need to change, although even pciu states and countries endpoints, which return data that doesn't really need protecting, both still fall under evss authorization because the headers require user attributes. :man_shrugging: I'm leaning towards removing the base controller either way 😃